### PR TITLE
chore(bindings): Move `crypto-nodejs` and `crypto-js` into the `bindings/` directory

### PR DESCRIPTION
**Pull request by Hywan**
This PR is the first step to move bindings into their own root, the `/bindings` directory (instead of `/crates`).

This is a first step because it mostly consists of a cleanup of the `.github/workflows/` files.

* The cleanup has started by creating a new `bindings_ci.yml` file (why this naming? because we can't have a subdirectory `bindings/ci.yml`…).
* Then, I've cleaned up the `ci.yml` file by removing everything related to bindings. I've continued by including the `wasm.yml` file inside `ci.yml` because, with the new organisation, `ci.yml` contains everything related to the `/crates` directory. I've taken the liberty to rename the jobs, so that names are consistent *_and_* short (the Github UI only proposes a narrow column, hint: `[m]` is short for `matrix-sdk`, cf the `[matrix]`'s logo)
* Then, I've updated the other workflow files just to provide uniformity amongst various namings.

Here is how it looks in [Github UI](https://github.com/matrix-org/matrix-rust-sdk/pull/781/checks):

* 🐧 stand for Linux
* 🍏 stands for macOS
* 🕸 stands for WebAssembly
* \[m\] stands for matrix-sdk (cf \[matrix\]'s logo)
* 🦀 stands for Rust

Screenshot 2022-06-22 at 15 15 01

And for the bindings:

Screenshot 2022-06-22 at 15 15 16

There is an addition here: `matrix-sdk-crypto-nodejs` was previously only tested on Linux, now it's tested on macOS too.

### CODE REVIEW

**Comment by jplatte** [comment]
Hm, what was wrong with the previous job names?

**Comment by Hywan** [comment]
> Hm, what was wrong with the previous job names?

It was too long and didn't fit in the Github UI. That was really annoying to find a particular job. Also, we had a lot of repetitive information, like “linux /” at the front, which could be replaced by a small one symbol wide, emoji.

File: .github/workflows/bindings_ci.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+
+env:
+  CARGO_TERM_COLOR: always
+  MATRIX_SDK_CRYPTO_NODEJS_PATH: bindings/matrix-sdk-crypto-nodejs
+  MATRIX_SDK_CRYPTO_JS_PATH: bindings/matrix-sdk-crypto-js
```
Comment (jplatte) [inline .github/workflows/bindings_ci.yml:18]: Why put this in environment variables?

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
-# xtask and labs should only be compiled when invoked explicitly
+members = [
+    "benchmarks",
+    "bindings/matrix-sdk-crypto-js",
+    "bindings/matrix-sdk-crypto-nodejs",
+    "crates/*",
+    "labs/*",
+    "xtask",
+]
+# xtask, labs and the bindings should only be invoked explicitly
```
Comment (jplatte) [inline Cargo.toml:10]: "compiled when" (or maybe "built when") should still be included, you can't really 'invoke' the bindings.

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-name: Docs
+name: Documentations
```
Comment (jplatte) [inline .github/workflows/docs.yml:1]: Should be "Documentation", idk why it's abbreviated "Docs" but it doesn't actually have a plural form in English ^^

File: .github/workflows/appservice.yml [hunk 4]
```diff
@@ -23,10 +23,13 @@ jobs:
     strategy:
       fail-fast: true
       matrix:
-        - name: 🐧
-          os: ubuntu-latest
-        - name: 🍏
-          os: macos-latest
+        os: [ubuntu-latest, macos-latest]
```
Comment (jplatte) [inline .github/workflows/appservice.yml:26]: This line is unnecessary if both entries are also listed in `include`.

File: .github/workflows/bindings_ci.yml (hunk 1 above)
Comment (Hywan) [reply .github/workflows/bindings_ci.yml:18]: To not repeat myself. Now that I'm thinking of it, we may want to declare variables, but not in the global environment… I'm checking the [documentation](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions) and I don't see another way to declare variables. Did I miss something?

File: .github/workflows/docs.yml (hunk 3 above)
Comment (Hywan) [reply .github/workflows/docs.yml:1]: Ah!
Fixed to “Documentation”, and I've renamed the file `documentation.yml`.

File: Cargo.toml (hunk 2 above)
Comment (Hywan) [reply Cargo.toml:10]: :+1:

File: .github/workflows/bindings_ci.yml (hunk 1 above)
Comment (Hywan) [reply .github/workflows/bindings_ci.yml:18]: I'm repeating these paths in several places, that's why I'm using a variable. But I agree that the size is not an argument here.

**Comment by Hywan** [comment]
I'm merging the PR as all jobs are green. I'm updating the branch protection rules right now.

**Pull request merged**